### DISCUSSION TITLE
How to use templ with pocketbase framework?

### DISCUSSION THREAD

**Discussion by heyarviind** [pocketbase/pocketbase#4129] [Q&A]
I am using Pocketbase as the framework in Golang with templ as the templateing system. I want to know if a user is authenticated in the `templ` template. Currently, I can think of passing the variables in each handler to the templ function but this doesn't seem like the best solution.

```
e.Router.GET("/dashboard", func(c echo.Context) error {
    return pages.DashboardPage(**<should I pass the context here>**).Render(c.Request().Context(), c.Response().Writer)
})
```

Is there any way i can directly get to know in the templ template? that would be so cool.

**Comment by heyarviind** [comment]
@ganigeorgiev can you please help me with this?

**Reply by ganigeorgiev** [reply]
There is no need to ping me, I receive notifications anyway for any new issue/discussion or comment. I didn't respond to your question because I don't understand what exactly you are asking and I don't think I can help you as I don't have much experience with `a-h/templ`.
This is probably not what you are looking for, but if you want to access the auth state you can read it from the `echo.Context` (see [Retrieving current auth state](https://pocketbase.io/docs/go-routing/#retrieving-the-current-auth-state)). Whether you'll pass it as an argument or some base/common struct to your templ component depends on your preference.
